### PR TITLE
Unify param names

### PR DESCRIPTION
**Pull request by DrLachie**
Deals with issue #73 

I've changed psf_num_iter to decon_num_iter. 
I've also left in the CLI_PARAM_MAP as it deals with things like roi_list going inside the crop parameter for example. Tested a bunch of permutations and seems to behave as expected.

### CODE REVIEW

**Comment by DrLachie** [comment]
Going to need to change the tests as the cli and decon tests will fail for the new parameter names

**Comment by multimeric** [comment]
Nice, this is much cleaner. I can see that you also actioned the change to `Optional[Tuple]` now that it has been fixed upstream. 

Can you also pull from master to simplify the diff please.

File: core/lls_core/cmds/__main__.py [hunk 1]
```diff
@@ -144,30 +141,25 @@ def process(
...
-    roi_indices: List[int] = field_from_model(CropParams, "roi_subset"),
-    # Ideally this and other range values would be defined as Tuples, but these seem to be broken: https://github.com/tiangolo/typer/discussions/667
-    z_start: Optional[int] = Option(0, help="The index of the first Z slice to use. All prior Z slices will be discarded.", show_default=False),
-    z_end: Optional[int] = Option(None, help="The index of the last Z slice to use. The selected index and all subsequent Z slices will be discarded. Defaults to the last z index of the image.", show_default=False),
-
+    roi_list: List[Path] = field_from_model(CropParams, "roi_list", description="A list of paths pointing to regions of interest to crop to, in ImageJ format."), #Option([], help="A list of paths pointing to regions of interest to crop to, in ImageJ format."),
+    roi_subset: List[int] = field_from_model(CropParams, "roi_subset"),
+    z_range: Optional[Tuple[int,int]] = Option(None, help="The index of the first Z slice to use. All prior Z slices will be discarded.", show_default=False),
```
Comment (multimeric) [inline core/lls_core/cmds/__main__.py:152]:
```suggestion
    z_range: Optional[Tuple[int,int]] = field_from_model(CropParams, "z_range"),
```

Comment (pr4deepr) [reply core/lls_core/cmds/__main__.py:152]: Documentation should say the index of first and last slice slice, right?
@DrLachie

Comment (DrLachie) [reply core/lls_core/cmds/__main__.py:152]: Yeah that's correct - I'll double check the help descriptions and update.

Comment (DrLachie) [reply core/lls_core/cmds/__main__.py]: I've updated descriptions  - I also added an extra logging line that includes the filename to be processed which is handy sometimes

**Comment by multimeric** [comment]
Looks good. Just noting that the `field_from_model` function I created copies the help from the Pydantic model if it exists. If the CLI usage and the Python usage are the same, I prefer to keep the description in one place. For example for devonvolution iterations: https://github.com/BioimageAnalysisCoreWEHI/napari_lattice/blob/f57b4895fe0479e4fb35cd78a68484dd24445c38/core/lls_core/models/deconvolution.py#L25-L28

The `description=` argument is for the arguments that behave differently in the CLI, like `physical_pixel_sizes`, because it's helpful to explain to the user how to pass a tuple via the CLI. For the Python API this is self-evident.

**Comment by multimeric** [comment]
@DrLachie

**Comment by DrLachie** [comment]
Just so I'm clear, I remove the descirptions from the options in __main__ and have it pull those from the models which is the preferable place for them? 

I see some inputs have a help rather than descirption which doesn't behave the same way?

**Comment by multimeric** [comment]
Yes that's right. So the `field_from_model` tries to convert a Pydantic `Field` into a CLI `Option`. 

* `field_from_model(...)` copies a model field and uses its description verbatim
* `field_from_model(..., description=)` lets you copy the field but change the description to make it CLI specific
* `field_from_model(..., extra_description=)` lets you append CLI-specific help in addition to the field's description
* `Option(help=)` doesn't copy from anywhere. The reason I used this in the past is because we had some CLI args like `z_start` that didn't correspond to a model field.

**Comment by multimeric** [comment]
Closing as #82 has an updated version of this.

**Pull request closed without merging**